### PR TITLE
Add complete HCL2 examples + allow to name a singular source block

### PR DESCRIPTION
**Pull request by azr**
Hello there

This is an HCL2 basic config based from bento boxes. I really like the degree of reusability HCL2 brings to the code !
~I'm opening this as a draft because I'm unsure wether it's actually too big for an example.~ It seems like it is okay like this.

This also adds the possibility to name a builder from a singular block code which made the example way better :

```hcl
source "type" "a" {}

build {
  name = "this_build"

  source "source.type.a" {
    name = "1604"
    # config for ubuntu 1604
  }
  source "source.type.a" {
    name = "1804"
    # config for ubuntu 1804
  }
}
```
So here we can easily differentiate ubuntu 16.04 from ubuntu 18.04 since they are based on the same common source.

### CODE REVIEW

File: examples/hcl/ubuntu/build.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    boot_wait               = "10s"
+  }
+
+  source "source.virtualbox-iso.base-ubuntu-amd64" {
+    iso_url                 = local.iso_url_ubuntu_1804
+    iso_checksum            = "file:${local.iso_checksum_url_ubuntu_1804}"
+    output_directory        = "ubuntu_1804_amd64"
+    boot_command            = local.ubuntu_1804_boot_command
+    boot_wait               = "5s"
+  }
```
Comment (azr) [inline examples/hcl/ubuntu/build.pkr.hcl:26]: This is really nice but both builds will be named `ubuntu.virtualbox-iso.base-ubuntu-amd64` and therefore you cannot only build ubuntu 1804 or ubuntu 1604, which I think would be nice to have here.

Comment (azr) [reply examples/hcl/ubuntu/build.pkr.hcl]: Ah never-mind I added the possibility to name source there

**Comment by azr** [comment]
I just added the possibility to name a singular block, this makes these examples way better

File: examples/hcl/ubuntu/build.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+build {
+  name = "ubuntu"
+  description = <<EOF
+This build creates ubuntu images for ubuntu versions :
+* 16.04
+* 18.04 
+For the following builers :
+* virtualbox-iso
+* parallels-iso
+* vmware-iso
```
Comment (azr) [inline examples/hcl/ubuntu/build.pkr.hcl:11]:
```suggestion
* vmware-iso
* qemu
* vspher-iso
```

File: examples/hcl/ubuntu/build.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+* qemu
+* vspher-iso
```
Comment (azr) [inline examples/hcl/ubuntu/build.pkr.hcl:13]:
```suggestion
* vsphere-iso
```

File: examples/hcl/ubuntu/build.pkr.hcl [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    vm_name                 = "ubuntu-16.04"
+    iso_url                 = local.iso_url_ubuntu_1604
+    iso_checksum            = "file:${local.iso_checksum_url_ubuntu_1604}"
+  }
+
+  provisioner "shell" {
+    environment_vars  = [ "HOME_DIR=/home/vagrant" ]
+    execute_command   = "echo 'vagrant' | {{.Vars}} sudo -S -E sh -eux '{{.Path}}'"
+    expect_disconnect = true
+    scripts           = fileset(".", "etc/scripts/*.sh")
```
Comment (azr) [inline examples/hcl/ubuntu/build.pkr.hcl:96]: TODO: document here

File: examples/hcl/ubuntu/build.pkr.hcl [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    boot_command            = local.ubuntu_1804_boot_command
+  }
+
+  source "source.qemu.base-ubuntu-amd64" {
+    name                    = "16.04"
+    iso_url                 = local.iso_url_ubuntu_1604
+    iso_checksum            = "file:${local.iso_checksum_url_ubuntu_1604}"
+    output_directory        = "qemu_iso_ubuntu_1604_amd64"
+    boot_command            = local.ubuntu_1604_boot_command
+  }
```
Comment (azr) [inline examples/hcl/ubuntu/build.pkr.hcl:75]: It would be nice to be able to link/reference these from the docs. 🤔

Comment (SwampDragons) [reply examples/hcl/ubuntu/build.pkr.hcl:75]: IT IS SO MUCH SMALLER AND CLEANER <3 <3 <3

Comment (azr) [reply examples/hcl/ubuntu/build.pkr.hcl:75]: THANKS ! I AGREE :D HCL2 IS SO NICE :D

**Comment by azr** [comment]
Okay I think this is good to go.

**Pull request merged**